jena-base/src/main/java/org/apache/jena/atlas/io/IO.java

```diff
@@ -441,13 +441,50 @@ public static String readWholeFileAsUTF8(InputStream in) {
         }
     }
 
+    /** Read a given number of characters from a stream as UTF-8.
+     * Append a given marker such as "..." if more data remains in the stream.
+     * Closing the stream is the caller's responsibility.
+     *
+     * @param in           InputStream to be read
+     * @param maxWidth     Maximum number of bytes (rather than characters) to read from the input stream
+     * @param abbrevMarker A string to append to the result if the input stream was not fully consumed
+     * @return      String
+     */
+    public static String abbreviateAsUTF8(InputStream in, int maxWidth, String abbrevMarker) throws IOException {
+        return abbreviate(in, StandardCharsets.UTF_8, maxWidth, abbrevMarker);
+    }
+
+    /** Read a given number of characters from a stream as UTF-8.
+     * Append a given marker such as "..." if more data remains in the stream.
+     * Closing the stream is the caller's responsibility.
+     */
+    public static String abbreviate(InputStream in, Charset charset, int maxWidth, String abbrevMarker) throws IOException {
+        return abbreviate(new InputStreamReader(in, charset), maxWidth, abbrevMarker);
+    }
+
+    /** Read a given number of characters from a stream as UTF-8.
+     * Append a given marker such as "..." if more data remains in the stream.
+     * Closing the stream is the caller's responsibility.
+     */
+    public static String abbreviate(Reader reader, int maxWidth, String abbrevMarker) throws IOException {
+        char[] buffer = new char[maxWidth + 1];
+        int n = IOUtils.read(reader, buffer);
+        StringBuilder sb = new StringBuilder();
+        if (n > 0) {
```

Comment:
Minor: The "if" test isn't needed.

If there is a test, it could be:
```
    int n = IOUtils.read(reader, buffer);
    if ( n == 0 )
       return "";
```

Reply:
Removed the if statement (I was incorrectly assuming it could return -1 like the read methods on InputStream)